sherpa/astro/models/tests/test_astro_model.py

```diff
@@ -102,6 +102,35 @@ def test_send_keyword_1d(cls):
     # Not guaranteed to produce interesting results
     x = [10, 12, 13, 15]
     y1 = mdl(x)
+    assert len(mdl._cache) == 1
+    y2 = mdl(x, not_an_argument=True)
+    assert len(mdl._cache) == 2
+    assert y2 == pytest.approx(y1)
+
+
+@pytest.mark.parametrize("cls", [models.Atten,
+                                 models.BBody,
+                                 models.BBodyFreq,
+                                 models.BPL1D,
+                                 models.Beta1D,
+                                 models.Edge,
+                                 models.LineBroad,
+                                 models.Lorentz1D,
+                                 models.NormBeta1D,
+                                 models.PseudoVoigt1D,
+                                 models.Schechter,
+                                 models.Voigt1D])
+def test_send_keyword_1d_no_cache(cls):
+    """What happens if we use an un-supported keyword and disable caching?"""
+
+    mdl = cls()
+    mdl.cache = 0
+    if cls == models.LineBroad:
+        mdl.vsini = 1e6
+
+    # Not guaranteed to produce interesting results
+    x = [10, 11.3, 13, 15]
+    y1 = mdl(x)
```

Comment:
I'm too lazy to check: do we have a test that after `mdl.cache = 0` a call to `mdl(x)` we still have `len(mdl._cache) == 0`?

Reply:
yes: https://github.com/sherpa/sherpa/pull/2166/files#diff-1426e384ca73b9afdca1ae2b6092d7f22f29e67ceef6035c10119eb0f6310592R945-R958

Reply:
ta.